src/test/regress/expected/multi_with.out

```diff
@@ -0,0 +1,870 @@
+                  events_table
+              WHERE
+                  users_table.user_id = events_table.user_id
+              GROUP BY
+                  users_table.user_id,
+                  events_table.event_type
+            
+            )
+            SELECT 
+                uid, event_type, value_2, value_3
+            FROM
+               (
+                (
+                  SELECT 
+                    user_id as uid
+                  FROM 
+                    users_events 
+                ) users
+                join
+                  events_table
+                on
+                  users.uid = events_table.event_type
+                ) a
+            )
+          SELECT 
+            * 
+          FROM 
+            users_events
+          ORDER BY
+            1, 3, 2, 4
+          LIMIT 100
+        )
+        SELECT 
+          * 
+        FROM 
+          users_events
+        LIMIT 90
+      )
+      SELECT 
+        * 
+      FROM 
+        users_events
+      LIMIT 50
+    )
+    SELECT 
+      uid, event_type, value_2, sum(value_3) as sum_3
+    FROM 
+      users_events
+    GROUP BY
+      1, 2, 3
+    LIMIT 40
+  )
+  SELECT 
+    uid, event_type, sum(value_2) as sum_2, sum(sum_3) as sum_3
+  FROM 
+    users_events
+  GROUP BY
+    1, 2
+  LIMIT 30   
+)
+SELECT 
+  uid, avg(event_type), sum(sum_2), sum(sum_3)
+FROM 
+  users_events
+GROUP BY
+  1;
+ uid |          avg           | sum | sum 
+-----+------------------------+-----+-----
+   1 | 1.00000000000000000000 |   3 |  72
+(1 row)
+
+-- Nested CTEs - joined with local table. Not supported yet.
+WITH users_events AS (
+  -- router select query
+  WITH users_events_1 AS (
+    SELECT
+      *
+    FROM
+      users_table
+    WHERE
+      user_id = 1
+  ),
+  -- real-time select query
+  users_events_2_3 AS (
+    SELECT
+      *
+    FROM
+      users_table
+    WHERE
+      user_id = 2
+      OR
+      user_id = 3
+  ),
+  -- router select query
+  -- sub CTE is a real-time executor query but the top level is router select
+  users_events_4 AS (
+    WITH users_events_4_5 AS (
+    SELECT
+      *
+    FROM
+      users_table
+    WHERE
+      user_id = 4
+      OR
+      user_id = 5
+    )
+    SELECT
+        *
+    FROM
+        users_events_4_5
+    WHERE
+        user_id = 4
+  ),
+  -- merge all the results from CTEs
+  merged_users AS (
+      SELECT
+        *
+      FROM
+        users_events_1
+    UNION
+      SELECT
+        *
+      FROM
+        users_events_2_3
+    UNION 
+      SELECT
+        *
+      FROM
+        users_events_4
+  ),
+  -- join with a local table 
+  local_users AS (
+    SELECT
+      *
+    FROM
+      (SELECT user_id as uid
+      FROM local_users) l_users
+      join
+      merged_users
+      on (l_users.uid = merged_users.user_id)
+  )
+  SELECT
+    *
+  FROM
+    local_users
+)
+SELECT 
+  * 
+FROM
+  users_events
+ORDER BY
+  1, 2, 3, 4, 5, 6
+LIMIT
+  20;
+ERROR:  relation local_users is not distributed
```

Comment:
Might be worth also having a test case in which `local_users` is queried in a CTE by itself, since that is supported.